Source/WebCore/platform/mediarecorder/cocoa/AudioSampleBufferCompressor.mm

```diff
@@ -534,12 +534,9 @@
 
 void AudioSampleBufferCompressor::addSampleBuffer(CMSampleBufferRef buffer)
 {
-    // Heap allocations are forbidden on the audio thread for performance reasons so we need to
-    // explicitly allow the following allocation(s).
-    DisableMallocRestrictionsForCurrentThreadScope disableMallocRestrictions;
-    m_serialDispatchQueue->dispatchSync([this, buffer] {
+    m_serialDispatchQueue->dispatch([this, buffer = RetainPtr { buffer }] {
```

Comment:
There is no guarantee anymore that `this` stays alive given flush may not be called before its destructor.
I guess we could ensure that it stays valid by having a dispatchSync in its destructor.
The regular approach would probably be to make AudioSampleBufferCompressor thread safe ref counted.

Reply:
doh good point